### PR TITLE
Separate demo and production builds

### PR DESCRIPTION
**Pull request by justinvdm**
This will help, since having the dummy api and demo code in the production build is unnecessary, unused bloat. It also helps us control how the app is run (for eg, whether it should use the dummy ureport api or the actual ureport api).

### CODE REVIEW

**Comment by justinvdm** [comment]
@hodgestar ready for review.

File: paths.js [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+
+src.all = [
+    'src/**/*.js'
+];
+
+module.exports = {
+    src: src,
+    dest: {
+        prd: 'lib/vumi-ureport.js',
+        demo: 'lib/vumi-ureport.demo.js'
```
Comment (hodgestar) [inline paths.js]: Bikeshed: I wonder if we shouldn't put these outside 'lib' so that it's clear which files are autogenerated?

**Comment by hodgestar** [comment]
Other than the minor bikeshed, :+1:.

**Comment by hodgestar** [comment]
Actually, let me retract my bikeshed -- it'd likely require too much fiddling with index.json and test files. Let's leave it for now.

File: paths.js (hunk 1 above)
Comment (justinvdm) [inline paths.js]: Sounds good, will do.

**Comment by justinvdm** [comment]
@hodgestar Na, it makes to put it outside of `lib/`. Besides, the commit's diffsize was 4 :) Ready for review.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**